hazelcast-sql/src/main/java/com/hazelcast/sql/impl/expression/FieldAccessExpression.java

```diff
@@ -76,33 +76,31 @@ public T eval(Row row, ExpressionEvalContext context) {
     @Override
     public T eval(final Row row, final ExpressionEvalContext context, boolean useLazyDeserialization) {
         // Use lazy deserialization for nested queries. Only the last access should be eager.
-        final Object res = ref.eval(row, context, true);
-        if (res == null) {
+        final Object result = ref.eval(row, context, true);
+        if (result == null) {
             return null;
         }
 
-        if (isPrimitive(res.getClass())) {
+        if (isPrimitive(result.getClass())) {
             throw QueryException.error("Field Access expression can not be applied to primitive types");
         }
 
-        if (getterCache.get() == null) {
-            getterCache.compareAndSet(null, new EvictableGetterCache(
+        if (getterCache == null) {
+            getterCache = new EvictableGetterCache(
                     MAX_CLASS_COUNT,
                     MAX_GETTER_PER_CLASS_COUNT,
                     GetterCache.EVICTABLE_CACHE_EVICTION_PERCENTAGE,
                     false
-            ));
+            );
         }
 
-        // defensive check, should never happen
-        final GetterCache cache = getterCache.get();
-        assert cache != null : "GetterCache should never be null";
-
-        final Extractors extractors = Extractors.newBuilder(context.getSerializationService())
-                .setGetterCacheSupplier(() -> cache)
-                .build();
         try {
-            return (T) type.convert(extractors.extract(res, name, useLazyDeserialization));
+            Object value = result instanceof GenericRecord
```

Comment:
unqualified reference to `GenericRecord` is misleading, we also have our own `GenericRecord` interface.

Reply:
What will we gain if we write it like the following? Isn't it obvious that `GenericRecord` is Avro by looking at the positive side of ternary operator? (I wish Java had an `import x as y` statement.)
```java
Object value = result instanceof org.apache.avro.generic.GenericRecord
        ? AvroQueryTarget.extractValue((org.apache.avro.generic.GenericRecord) result, name)
```

Reply:
> Isn't it obvious that GenericRecord is Avro by looking at the positive side of ternary operator

maybe, but when I was first reading it I was puzzled for a while. But ok, we can keep it as is.